services/director-v2/tests/unit/test_modules_dynamic_sidecar_docker_api.py

```diff
@@ -63,27 +60,31 @@ def network_config(simcore_services_network_name: str) -> Dict[str, Any]:
 
 @pytest.fixture
 async def ensure_swarm_network(
-    loop: BaseEventLoop, network_config: Dict[str, Any]
+    loop: BaseEventLoop,
+    network_config: Dict[str, Any],
+    docker_client: aiodocker.docker.Docker,
+    async_docker_swarm: None,
 ) -> None:
     network_id = None
-    async with docker_client() as client:
-        try:
-            network_id = await create_network(network_config)
-            yield
-        finally:
-            if network_id is not None:
-                docker_network = await client.networks.get(network_id)
-                assert await docker_network.delete() is True
+    try:
+        network_id = await docker_api.create_network(network_config)
+        yield
+    finally:
+        if network_id is not None:
+            docker_network = await docker_client.networks.get(network_id)
+            assert await docker_network.delete() is True
```

Comment:
question, why is this not done when the docker swarm is shutoff?
Also, this will delete whatever is on the host machine. even if it was not started by a test.

Could you not list what was available *before* creating the swarm. then delete whatever was created after that moment?

Reply:
The network is created on the fly from the configuration. Also the `network_id` is used to delete the network. A network id will only be returned if the network was created (or did I misunderstand this).

So basically the network is only deleted if it is created.